lib/RoundRobinSet.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright © 2013 - 2017 Tinder, Inc.
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+'use strict';
+
+module.exports = class RoundRobinSet {
+  
+  constructor(arr) {
+    const newArr = arr.slice(0);
+
+    this._current = 0;
+
+    for (let i = 0, l = newArr.length; i < l; i++)
+      newArr[i]._rrindex = i;
+
+    this.items = newArr;
+  }
+
+  obtain() {
```

Comment:
How do you think we should handle that? I'd think either return null, or put an assertion into the constructor to prevent it from ever being possible.